env-files/torch/generic_torch.sh

```diff
@@ -155,19 +159,28 @@ else
   # Cleaner Horovod installation
 	# https://github.com/horovod/horovod/pull/3998
   # Assume that Horovod env vars are already in the current env!
-  pip install --no-cache-dir git+https://github.com/thomas-bouvier/horovod.git@compile-cpp17
+  pip install --no-cache-dir git+https://github.com/thomas-bouvier/horovod.git@compile-cpp17 || exit 1
 fi
 
 # get required libraries in reqs.txt
-if [ -f "${cDir}/$ENV_NAME/lib/python${pver}/site-packages/torchnlp/_third_party/weighted_random_sampler.py" ]; then
-   echo 'required libs already exist'
-else
+# if [ -f "${cDir}/$ENV_NAME/lib/python${pver}/site-packages/torchnlp/_third_party/weighted_random_sampler.py" ]; then
+   # echo 'required libs already exist'
+# else
 #   pip install -r Scripts/reqs.txt --no-cache-dir
 
   # fix int bug: modify l.4 of /torchnlp/_third_party/weighted_random_sampler.py
-  var='int_classes = int'
-  sed -i "4s|.*|$var|" \
-    ${cDir}/$ENV_NAME/lib/python${pver}/site-packages/torchnlp/_third_party/weighted_random_sampler.py
+  # var='int_classes = int'
+  # sed -i .backup_file "4s|.*|$var|" \
+    # ${cDir}/$ENV_NAME/lib/python${pver}/site-packages/torchnlp/_third_party/weighted_random_sampler.py || exit 1
+  # Deleting unnecessary backup file
+  # rm ${cDir}/$ENV_NAME/lib/python${pver}/site-packages/torchnlp/_third_party/weighted_random_sampler.py.backup_file
+# fi
+
+# Install Pov4ML
+if [[ "$OSTYPE" =~ ^darwin ]] ; then
+  pip install "prov4ml[apple]@git+https://github.com/matbun/ProvML@main" || exit 1
```

Comment:
To be consistent with the other `pip install` commands in this script, please add `--no-cache-dir` option